osbenchmark/builder/installers/opensearch_installer.py

```diff
@@ -0,0 +1,113 @@
+                    log_path=node_log_dir,
+                    heap_dump_path=node_heap_dump_dir,
+                    data_paths=None,
+                    telemetry=None)
+
+    def _prepare_node(self, host, node, binary, all_node_ips):
+        self._prepare_directories(host, node)
+        self._extract_opensearch(host, node, binary)
+        self._update_node_binary_path(node)
+        self._set_node_data_paths(node)
+        # we need to immediately delete the prebundled config files as plugins may copy their configuration during installation.
+        self._delete_prebundled_config_files(host, node)
+        self._prepare_config_files(host, node, all_node_ips)
+
+    def _prepare_directories(self, host, node):
+        directories_to_create = [node.binary_path, node.log_path, node.heap_dump_path]
+        for directory_to_create in directories_to_create:
+            self.path_manager.create_path(host, directory_to_create)
+
+    def _extract_opensearch(self, host, node, binary):
+        self.logger.info("Unzipping %s to %s", binary, node.binary_path)
+        self.executor.execute(host, "tar -xzvf {} --directory {}".format(binary, node.binary_path))
+
+    def _update_node_binary_path(self, node):
+        node.binary_path = os.path.join(node.binary_path, "opensearch*")
+
+    def _set_node_data_paths(self, node):
+        node.data_paths = [os.path.join(node.binary_path, "data")]
+
+    def _delete_prebundled_config_files(self, host, node):
+        config_path = os.path.join(node.binary_path, "config")
+        self.logger.info("Deleting pre-bundled OpenSearch configuration at [%s]", config_path)
+        self.path_manager.delete_path(host, config_path)
+
+    def _prepare_config_files(self, host, node, all_node_ips):
+        config_vars = self.get_config_vars(host, node, all_node_ips)
+        self.config_applier.apply_configs(host, node, self.provision_config_instance.config_paths, config_vars)
+
+    def get_config_vars(self, host, node, all_node_ips):
+        provisioner_defaults = {
+            "cluster_name": self.provision_config_instance.variables["cluster_name"],
+            "node_name": node.name,
+            "data_paths": node.data_paths[0],
+            "log_path": node.log_path,
+            "heap_dump_path": node.heap_dump_path,
+            # this is the node's IP address as specified by the user when invoking Benchmark
+            "node_ip": host.address,
+            # this is the IP address that the node will be bound to. Benchmark will bind to the node's IP address (but not to 0.0.0.0). The
+            "network_host": host.address,
+            "http_port": str(node.port),
+            "transport_port": str(node.port + 100),
+            "all_node_ips": "[\"%s\"]" % "\",\"".join(all_node_ips),
+            # at the moment we are strict and enforce that all nodes are master eligible nodes
+            "minimum_master_nodes": len(all_node_ips),
+            "install_root_path": node.binary_path
+        }
```

Comment:
Nit: I doubt there's going to be another iteration of an installer but how do you feel about adding these keys of provisioner_defaults to an enum? 

Not sure where else provisioner_defaults might be used in the code. It might make this dictionary a more muddled; However, I'm curious if we should start implementing enums throughout OSB to enable reusability of terms and if we were ever to update the naming conventions of variables such as these, it'd be easier. Curious to what others think too.

Reply:
Different installers have different keys for their default variables. There's some overlap, but I don't think there's a good solution for a rigid model representing the same.